tests/modal.ts

```diff
@@ -0,0 +1,1097 @@
+                <aalam-modal>
+                    <div slot="modal-body">
+                        <div style="height:20px;width:30px;background:grey">
+                        </div>
+                        <div style="height:40px;width:20px"></div>
+                    </div>
+                </aalam-modal>`);
+        el1.show();
+        const parent_el = el1.parentElement;
+        expect(body_el).to.equal(parent_el)
+    });
+
+    it('stack', async () => {
+        const el1 = await fixture (html`
+            <div id = "parent_holder">
+                <aalam-modal id="outer_modal">
+                    <div slot="modal-body">
+                        <div style="height:20px;width:30px;background:grey">
+                        </div>
+                        <div id="child_modal">
+                            <aalam-modal id="inner_modal">
+                                <div slot="modal-body">
+                                    <div style="background:orange"></div>
+                                </div>
+                            </aalam-modal>
+                        </div>
+                    </div>
+                </aalam-modal>
+                <aalam-modal id="stack_modal" stack>
+                    <div slot="modal-body"></div>
+                </aalam-modal>
+            </div>`);
+
+        const out_mod = el1.querySelector("#outer_modal");
+        const out_par = out_mod.parentElement;
+        out_mod.show();
+        expect(out_mod.parentElement).to.equal(document.body);
+        const in_mod = out_mod.querySelector("#inner_modal");
+        const in_par = in_mod.parentElement;
+        out_mod.animationDur = 0;
+        in_mod.show();
+        expect(out_mod.parentElement).to.equal(out_par);
+        expect(in_mod.parentElement).to.equal(document.body);
+
+        const stack_modal = el1.querySelector("#stack_modal");
+        out_mod.show();
+        stack_modal.show();
+        stack_modal.animationDur =0;
+        expect(stack_modal.parentElement).to.equal(out_mod.parentElement);
+        expect(stack_modal.parentElement).to.equal(document.body);
+        expect(out_mod.parentElement).to.equal(document.body);
+    });
+
+    it('closesel', async () => {
+        const el1 = await fixture (html`
+            <div id = "parent">
+                <button id="open_btn"></button>
+                <aalam-modal id="modal" closesel=".closing">
+                       <div id="slot" slot="modal-body">
+                           <button id = "close_btn"></button>
+                           <button id="new_button"></button>
+                       </div>
+                </aalam-modal>
+            </div>
+        `);
+        const modal = el1.querySelector("#modal");
+        modal.animationDur = 0;
+        const slot_id = el1.querySelector("#slot");
+        const modal_par = modal.parentElement;
+        modal.show();
+        const new_btn = modal.querySelector("#new_button");
```

Comment:
Click the button before adding `.closing'` class and check if the modal is not closed. ANd then add this class and then click and it should close the modal.